### DISCUSSION TITLE
migration to react query v5 - issues

### DISCUSSION THREAD

**Discussion by idbd1** [dotansimha/graphql-code-generator#9823] [Q&A]
I'm having several issues after migrating tanstack query to v5 version. Is there anything I might have missed in the codegen.ts? how can I fix the warnings in generated.ts and how to pass options to the query?

```
const currentUserQuery = useCurrentUserQuery({ enabled: !isLoggingIn });
```
`TS2322: Type boolean is not assignable to type never`

<img width="751" alt="image" src="https://github.com/dotansimha/graphql-code-generator/assets/31188583/e2ae3e6d-84cc-4104-af01-c76fbaa212dd">

codegen config

```
// @ts-ignore
import type { CodegenConfig } from '@graphql-codegen/cli';

const config: CodegenConfig = {
  overwrite: true,
  schema: '../schema.graphql',
  reactQueryVersion: 5,
  documents: 'src/**/*.graphql',
  generates: {
    'src/generated/generated.ts': {
      plugins: ['typescript', 'typescript-operations', 'typescript-react-query'],
      config: {
        fetcher: '../fetcher#fetcher',
        namingConvention: {
          enumValues: 'keep',
        },
      },
    },
  },
};

export default config;
```

```
 "dependencies": {
    "@graphql-codegen/cli": "5.0.0",
    "@graphql-codegen/typescript-operations": "4.0.1",
    "@graphql-codegen/typescript-react-query": "6.0.0",
    "@graphql-eslint/eslint-plugin": "3.20.1",
    "@graphql-typed-document-node/core": "3.2.0",
    "@tanstack/eslint-plugin-query": "4.34.1",
    "@tanstack/react-query": "5.17.19",
```

**Comment by sebpalluel** [comment]
You can add `reactQueryVersion: 5` here:
```
config: {
        fetcher: '../fetcher#fetcher',
        namingConvention: {
          enumValues: 'keep',
        },
        reactQueryVersion: 5
      },
```
 It's working for me with latest current version `"@graphql-codegen/typescript-react-query": "^6.1.0"`

**Reply by nolafs** [reply]
Does not work for me. ESlint will not allow me compile it.

 return useQuery<CtaQuery, TError, TData>(
      ['cta', variables],
      customFetcher<CtaQuery, CtaQueryVariables>(CtaDocument, variables),
      options
    )};

TS2554: Expected 1-2 arguments, but got 3
